### PR TITLE
added Sidebar and Filter Functionality

### PR DESCRIPTION
**Pull request by atharvabhanage02**
### Sidebar and Filter Functionality

Added Sidebar component and Filter Functionality to the products

Features : 

-  SortBy Price :  Low to High , High to Low
- Product Availability :  InStock and Fast Delivery Products
- Filter By Category (The categories mentioned are demo categories and will be changed further)
- Filter By Rating

### CODE REVIEW

File: src/Components/Sidebar/sidebar.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    display: flex;
+}
+/* Sidebar */
+.sidebar{
+    position: fixed;
+    flex-grow: 0;
+    top: 5rem;
+    height: 100%;
+    border-right: 2px solid black;
+    padding: 1.25rem;
```
Comment (Kavita539) [inline src/Components/Sidebar/sidebar.css:20]: Suggestion : For spacings you can make use of variables to have consistency throughout the code.

**Review by Kavita539 — APPROVED** [review]
Apart from the comment, everything else looks good (You can make the changes later on)

- The code is clean, readable.
- Folder structuring is done nicely.
- UI looks good and every functionality is working.

File: src/Components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        <div className="product-sort">
+          <h3 className="filter-feature">Sort</h3>
+          <div className="categories">
+            <label className="input-select">
+              <input
+                type="radio"
+                id="low-to-high"
+                name="SORT_BY"
+                value="LOW_TO_HIGH"
+                checked={state.sortBy !== "LOW_TO_HIGH" ? false : true}
```
Comment (JantuDeb) [inline src/Components/Sidebar/Sidebar.jsx]: Don't need to return `true` or `false` ,  `!==` operator doing the same

File: src/Components/Sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+              />
+              <span>Price - Low to High</span>
+            </label>
+            <label class="input-select">
+              <input
+                type="radio"
+                id="high-to-low"
+                name="SORT_BY"
+                value="HIGH_TO_LOW"
+                checked={state.sortBy !== "HIGH_TO_LOW" ? false : true}
```
Comment (JantuDeb) [inline src/Components/Sidebar/Sidebar.jsx]: Same here

File: src/Components/Sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        <h3 className="filter-feature">Check Availability</h3>
+        <div className="categories">
+          <label class="input-select">
+            <input
+              type="checkbox"
+              id="stockCheck"
+              name="stockCheck"
+              className="checkbox-input"
+              value="CHECK_STOCK"
+              checked={state.outOfStock ? true : false}
```
Comment (JantuDeb) [inline src/Components/Sidebar/Sidebar.jsx]: If outOfStock is a boolean value then `checked={state.outOfStock}` is fine

File: src/Components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            />
+            <span>Show Products in Stock</span>
+          </label>
+          <label class="input-select">
+            <input
+              type="checkbox"
+              id="stockfastDelivery"
+              className="checkbox-input"
+              value="CHECK_FAST_DELIVERY"
+              checked={state.fastDelivery ? true : false}
```
Comment (JantuDeb) [inline src/Components/Sidebar/Sidebar.jsx]: same here

File: src/Components/Sidebar/Sidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                onChange={(e) =>
+                  dispatch({
+                    type: "FILTER_BY_RATINGS",
+                    payload: e.target.value,
+                  })
+                }
+              />
+              <span>1 stars & above</span>
+            </label>
+          </div>
```
Comment (JantuDeb) [inline src/Components/Sidebar/Sidebar.jsx:179]: You can follow the same approach as category, using a `map()` function

**Review by JantuDeb — APPROVED** [review]
Overall the codes are structured nicely written, jut go through the comment I left

File: src/Components/Sidebar/Sidebar.jsx (hunk 2 above)
Comment (atharvabhanage02) [reply src/Components/Sidebar/Sidebar.jsx]: Yes , I'll do that 
Thanks for the suggestion

**Pull request merged**